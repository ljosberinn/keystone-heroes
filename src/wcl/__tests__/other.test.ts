import {
  filterProfessionEvents,
  filterPlayerDeathEvents,
} from "../queries/events/other";
import allEvents from "./fixtures/allEvents.json";

describe("other", () => {
  test("filterProfessionEvents", () => {
    expect(filterProfessionEvents(allEvents)).toMatchSnapshot();
  });

  test("filterPlayerDeathEvents", () => {
    // PTR log https://www.warcraftlogs.com/reports/J3WKacdjpntmLT7C#fight=3&type=damage-done
    const actorPlayerSet: Parameters<typeof filterPlayerDeathEvents>[1] =
      new Set([1, 2, 3, 5, 60]);

    const result = filterPlayerDeathEvents(allEvents, actorPlayerSet);

    expect(result).toHaveLength(23);
    expect(result).toMatchSnapshot();
  });
});
