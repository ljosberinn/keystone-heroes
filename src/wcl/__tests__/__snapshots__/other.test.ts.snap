// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`other filterPlayerDeathEvents 1`] = `
[
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 1,
    "timestamp": 3186763,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 2,
    "timestamp": 3331087,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 2,
    "timestamp": 3424154,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 1,
    "timestamp": 3436647,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 5,
    "timestamp": 3489087,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 3,
    "targetMarker": 7,
    "timestamp": 3506311,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 60,
    "timestamp": 3512625,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 1,
    "timestamp": 3514282,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 2,
    "timestamp": 3517463,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 5,
    "timestamp": 3534137,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 3,
    "targetMarker": 7,
    "timestamp": 3559079,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 2,
    "timestamp": 3560518,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 5,
    "timestamp": 3563937,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 1,
    "timestamp": 3566890,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 60,
    "timestamp": 3568140,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 3,
    "targetMarker": 7,
    "timestamp": 3573319,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 2,
    "timestamp": 3576106,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 5,
    "timestamp": 3793055,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 1,
    "timestamp": 3889770,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 3,
    "targetMarker": 7,
    "timestamp": 3910966,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 1,
    "timestamp": 4036935,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 1,
    "timestamp": 4273729,
    "type": "death",
  },
  {
    "abilityGameID": 0,
    "pin": "0",
    "sourceID": -1,
    "targetID": 5,
    "timestamp": 4634252,
    "type": "death",
  },
]
`;

exports[`other filterProfessionEvents 1`] = `
[
  {
    "abilityGameID": 345130,
    "sourceID": 5,
    "targetID": 103,
    "timestamp": 2826042,
    "type": "cast",
  },
  {
    "abilityGameID": 307195,
    "sourceID": 1859,
    "sourceMarker": 6,
    "targetID": 1859,
    "targetMarker": 6,
    "timestamp": 12294824,
    "type": "applybuff",
  },
  {
    "abilityGameID": 321422,
    "sourceID": 4,
    "targetID": 4,
    "timestamp": 12295325,
    "type": "applybuff",
  },
  {
    "abilityGameID": 321422,
    "sourceID": 1846,
    "sourceMarker": 2,
    "targetID": 1846,
    "targetMarker": 2,
    "timestamp": 12297167,
    "type": "applybuff",
  },
]
`;
